src/zk.py

```diff
@@ -186,26 +186,30 @@ def _write(self, path, data, need_lock=True):
             logging.error(event.exception)
         return not event.exception
 
-    def _init_lock(self, name, read_lock=False):
+    def _init_lock(self, name, read_lock=False, host_description=None):
+        if host_description is None:
+            host_description = helpers.get_hostname()
         path = self._path_prefix + name
         if read_lock:
-            lock = self._zk.ReadLock(path, helpers.get_hostname())
+            lock = self._zk.ReadLock(path, host_description)
         else:
-            lock = self._zk.Lock(path, helpers.get_hostname())
+            lock = self._zk.Lock(path, host_description)
```

Comment:
Sorry, but the same inconsistency as in https://github.com/yandex/pgconsul/pull/69#discussion_r2068343040 is still here.
We can init lock on some path with one `host_description`, and later try get it with `_get_lock` with another `host_description`, but will get lock instance with first `host_description`.

To fix this you need to rework `self._locks` map, making a tuple `(name, host_description)` key in it.

Reply:
Host description in one instance of pgconsul is constant, there is no such problem

Reply:
> Host description in one instance of pgconsul is constant, there is no such problem

Then it should be put it into `Zookeeper` class constructor, like I've suggested in https://github.com/yandex/pgconsul/pull/69#discussion_r2068343040

It's all about usability and predictability of this class. For example:
Given that a lock on path `path/to/lock` is held by `fqdn2`, if you call consecutively
```
zk.try_acquire_lock(path='path/to/lock', host_description='fqdn1')
zk.try_acquire_lock(path='path/to/lock', host_description='fqdn2')
```
then in second call the same lock object will be found in `_get_lock`, and it will work incorrectly with `host_description='fqdn1'`, and won't realize that lock is already held and whole call should return success instead of failure.